opendatahub-io/odh-dashboard
# Add pipeline versions table, refactor pipelines table and pipelines selector

JIRA:
https://issues.redhat.com/browse/RHOAIENG-263
https://issues.redhat.com/browse/RHOAIENG-264

## Description


The PR did the following things:
1. Refactor the global pipelines table, and change it to bulk selection, change the columns to get the versions number and updated date (the latest version date)

<img width="1724" alt="Screenshot 2023-12-14 at 4 39 55 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/d7235502-a6ff-4761-9e59-736c1a0ee535">

2. Add the pipeline versions table to the expandable section in the row of the pipeline, when it's more than 10 versions, you can click on the View More button to load more versions, you can also sort the versions by name and create time.

**NOTE: The view run Button link is not implemented yet, needs to be addressed when refactoring the pipeline runs page.**

<img width="1724" alt="Screenshot 2023-12-14 at 4 42 16 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/dd588160-0fa8-4ea8-8812-930fa320cdf3">

3. Refactor the pipeline and pipeline version selector to make it also use the server-side filtering and sorting to keep the behavior aligned with what we have now.

## How Has This Been Tested?



1. Go to the global pipelines page
2. Try to expand the pipelines to see the pipeline versions
3. Click on the version should be able to navigate you to the version details page
4. Try to upload new versions to the pipelines
5. Try to create more than 10 pipeline versions under a pipeline to test the view more function
6. Try to sort the pipelines table and the pipeline versions table
7. Try to delete all the versions in one pipeline to check this state
<img width="1724" alt="Screenshot 2023-12-14 at 4 47 30 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/677ca02d-3d3e-4ab5-bc75-a030eb749351">
8. Play around with the pipeline selectors, it should have better filtering and sorting now

## Test Impact


Update the test cases for the pipeline selector. Didn't create new tests, because it makes more sense to use e2e tests for this.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
